### PR TITLE
Add Order.service_types

### PR DESCRIPTION
**Pull request by marcofucci**
This
- adds the many2many field Order.service_types
- adds service_types to the creation endpoint
- adds service_types to the update endpoint

As a service type can be disabled at any point, the endpoint raises a validation error if the client uses a value that was disabled at the time of creation of an order.

Disabled service types can be replaced by other service types if they were not disabled when the order was first created.

### CODE REVIEW

**Comment by marcofucci** [comment]
@reupen @elcct maybe the `disabled_on` field would actually be useful for other metadata models as well?

File: datahub/omis/order/test/factories.py [hunk 1]
```diff
@@ -19,6 +21,20 @@ class OrderFactory(factory.django.DjangoModelFactory):
...
+        if not create:
+            return  # Simple build, do nothing.
+
+        if extracted:
+            service_types = extracted
+        else:
+            service_types = ServiceType.objects.filter(disabled_on__isnull=True).order_by('?')[:2]
+
+        for service_type in service_types:
+            self.service_types.add(service_type)
```
Comment (reupen) [inline datahub/omis/order/test/factories.py]: `self.service_types.set(service_types)` should work.

There is a to_many_field decorator in datahub.core.test.factories which you could consider extending.

**Comment by reupen** [comment]
> @reupen @elcct maybe the disabled_on field would actually be useful for other metadata models as well?

Yes, it might be useful (particularly with the data migration) and it looks like a good way of handling this. Not sure if we should rush into adding it everywhere though, there's the front-end aspect to consider too.

File: datahub/omis/order/test/factories.py (hunk 1 above)
Comment (marcofucci) [inline datahub/omis/order/test/factories.py]: @reupen Have a look at my last commit and see if you like how I extended the `to_many_field`.

File: datahub/core/test/factories.py [hunk 2]
```diff
@@ -19,6 +19,19 @@ def to_many_field(wrapped_func):
 
     This decorator doesn't call the decorated function; the decorated function is only used
```
Comment (reupen) [inline datahub/core/test/factories.py]: Comment here needs updating.

File: datahub/core/test/factories.py [hunk 3]
```diff
@@ -28,8 +41,13 @@ def wrapping_func(self, create, extracted, **kwargs):
             # should not be set
             return
 
+        field = getattr(self, wrapped_func.__name__)
         if extracted:
```
Comment (reupen) [inline datahub/core/test/factories.py]: Should it be `if extracted is not None:`? So you can still pass an empty list etc.?

File: datahub/omis/order/test/factories.py [hunk 4]
```diff
@@ -19,6 +22,15 @@ class OrderFactory(factory.django.DjangoModelFactory):
...
     sector_id = Sector.aerospace_assembly_aircraft.value.id
 
+    @to_many_field
+    def service_types(self):
+        """
+        Add support for setting service_types.
+        If nothing specified when instantiating the object, the value returned by
+        this method will be used by default.
+        """
+        return ServiceType.objects.filter(disabled_on__isnull=True).order_by('?')[:2]
```
Comment (elcct) [inline datahub/omis/order/test/factories.py]: I thought selecting records in random order this way is a bit inefficient or is the table too small to make a difference?

File: datahub/core/test/factories.py (hunk 3 above)
Comment (marcofucci) [inline datahub/core/test/factories.py]: Yes I was thinking the same.

File: datahub/omis/order/test/factories.py (hunk 4 above)
Comment (marcofucci) [inline datahub/omis/order/test/factories.py]: Yeah the table is very small and it's just during tests so I'm not really worried.

**Comment by marcofucci** [comment]
@reupen @elcct have another quick look

File: datahub/omis/order/test/factories.py (hunk 4 above)
Comment (elcct) [inline datahub/omis/order/test/factories.py]: Great

**Pull request merged**